core/src/main/scala/kafka/consumer/ZookeeperConsumerConnector.scala

```diff
@@ -421,6 +421,12 @@ private[kafka] class ZookeeperConsumerConnector(val config: ConsumerConfig,
         Thread.sleep(config.offsetsChannelBackoffMs)
       }
     }
+ 
```

Comment:
Since we set lastCommittedPartitionsAndOffsets in any case, can you also remove line 354 and 371?

Reply:
Discussed offline, move the setting of lastCommittedPartitionAndOffsets outside of the while loop to make sure it will reflect the status of commit.